src/react-vite/08-stateful-hooks/stateful-hooks.md

```diff
@@ -2,51 +2,117 @@
 @parent learn-react-vite 8
 @outline 3
 
-@description TODO
+@description Work with React's useState hook to manage a component's state.
 
 @body
 
 ## Overview
 
-TODO
+When users make choices that need to be maintained or that affect other parts of
+the UI, we need to use "state." This lesson introduces React hooks that manage
+component state.
+
+## Objective: Manage component state using hooks
+
+React provides hooks that simplify managing state in a component. We will use
+some of these hooks to store the user's selections for state and city.
+
+> Note: Hooks must always appear in the body of the component **before** any
+> statements that might change the component's behavior. Place hooks at the top
+> of a component's body.
+
+### useState
+
+We can store state that persists through component rendering with the `useState`
+hook. You can set the initial state value when the component **first** renders
+by providing the value as an argument to the hook. _If you do not provide a
+value the initial state value will be `undefined`._
+
+This example shows a `useState` hook being set with an initial value from an
+`initialText` prop.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
+  const [value, setValue] = useState(initialText)
+}
+```
+
+As you can see in the previous example, `useState` returns an array with two
+elements: the first is the current value of the hook, and the second is a setter
+function that is used to update the state value. In the following code, the
+value and setter are being used to update changes to an input element.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
+  const [value, setValue] = useState(initialText)
+  return <input onChange={(evt) => setValue(evt.target.value)} value={value} /> 
+}
+```
+
+Every time a `useState`'s setter is invoked with a new value, React compares the
+new value with the current value. If the values are the same, nothing happens;
+**if the values are different, React will rerender the component** so the new
+value can be used to update the component. In the example above, when the user
+types or changes the value of the input, the `FancyInput` component is rendered
+again, and the input is updated with the current value.
+
+> Watch out! To determine if a state value has changed React compares value
+> types like `object` and array by instance, not by their contents. Other types,
+> like number, string, and boolean, are compared by their value.
+
+### useId
+
+One of the newer hooks that React includes is `useId`. Since the value of every
+`id` attribute in an HTML document must be unique, this hook is useful in
+creating a unique identifier string that can be used as the value of `id` props.
+
+```jsx
+const FancyInput: React.FC<Props> = ({ initialText }) => {
```

Comment:
If you can, try to ground these examples in a real use case (instead of `FancyInput`, a real potential component).

Reply:
done